### PR TITLE
[3.8] Vert.x 4.5.7 update

### PR DESCRIPTION
**Pull request by cescoffier**

### CODE REVIEW

**Comment by cescoffier** [comment]
I'm not sure; I had to do specific things and didn't have to add a lot of stuff to that PR.

**Comment by cescoffier** [comment]
Added another look to the backport branch, it seems OK - the 3.2 backport is way more tricky. 

I ran my tests against the 6 versions of GraalVM we "support" (22 EE / CE, 21 EE / CE, 17 EE / CE).

**Comment by cescoffier** [comment]
@gsmet BTW, your PR had a runtime init that should not be necessary in the NettyProcessor. The JdkSslClientContext does not lead to a problematic build time initialization.

File: bom/application/pom.xml [hunk 1]
```diff
@@ -120,9 +120,9 @@
         <slf4j-jboss-logmanager.version>2.0.0.Final</slf4j-jboss-logmanager.version>
         <wildfly-common.version>1.7.0.Final</wildfly-common.version>
         <wildfly-client-config.version>1.0.1.Final</wildfly-client-config.version>
-        <wildfly-elytron.version>2.2.3.Final</wildfly-elytron.version>
-        <jboss-threads.version>3.5.1.Final</jboss-threads.version>
-        <vertx.version>4.5.4</vertx.version>
+        <wildfly-elytron.version>2.3.1.Final</wildfly-elytron.version>
+        <jboss-threads.version>3.6.0.Final</jboss-threads.version>
```
Comment (gsmet) [inline bom/application/pom.xml:124]: Are these upgrades really necessary? They are not in my PR but I'm not sure they are?

File: independent-projects/resteasy-reactive/pom.xml [hunk 2]
```diff
@@ -60,8 +60,8 @@
         <version.enforcer.plugin>3.2.1</version.enforcer.plugin>
         <version.surefire.plugin>3.2.5</version.surefire.plugin>
         <mutiny.version>2.5.8</mutiny.version>
-        <smallrye-common.version>2.1.2</smallrye-common.version>
-        <vertx.version>4.5.4</vertx.version>
+        <smallrye-common.version>2.3.0</smallrye-common.version>
```
Comment (gsmet) [inline independent-projects/resteasy-reactive/pom.xml:63]: Is this strictly necessary? I'm not sure it's in line with what we have in the BOM so this is problematic.

**Review by gsmet — CHANGES_REQUESTED** [review]
> BTW, your PR had a runtime init that should not be necessary in the NettyProcessor. The JdkSslClientContext does not lead to a problematic build time initialization.

I didn't invent it. It was part of the PRs I backported.
I would prefer we follow the usual process except if strictly necessary as diverging might cause unnecessary issues.

I had no issue backporting the PRs so that's what I did. I added several comments inline with questions as this PR has more changes than what I actually backported and I don't know if there are good reasons.

**Comment by cescoffier** [comment]
@gsmet The problem is that the thing you backported has not being tested on all the variants and is likely going to break. Let's sync up tomorrow.

**Comment by cescoffier** [comment]
The regular backport is good enough.

**Pull request closed without merging**